tests/integration/security/authorization_test.go

```diff
@@ -1237,3 +1243,133 @@ func TestAuthorization_Audit(t *testing.T) {
 			rbacUtil.RunRBACTest(t, cases)
 		})
 }
+
+func configureExtensionProviders(t *testing.T, ctx framework.TestContext, ns string) {
+	// Get the existing mesh config.
+	systemNs := istio.ClaimSystemNamespaceOrFail(t, ctx)
+	configMap, err := ctx.Clusters().Default().CoreV1().ConfigMaps(systemNs.Name()).Get(context.Background(), "istio", v1.GetOptions{})
+	if err != nil {
+		ctx.Fatalf("Failed to get config map: %v", err)
+	}
+	meshConfig, err := mesh.ReadConfigMap(configMap, "mesh")
+	if err != nil {
+		ctx.Fatalf("Failed to read config map: %v", err)
+	}
+
+	// Create two extension providers using the same ext-authz service, one for HTTP and the other for gRPC ext-authz API.
+	service := fmt.Sprintf("%s/ext-authz", ns)
+	httpProvider := &v1alpha1.MeshConfig_ExtensionProvider{
+		Name: "ext-authz-http",
+		Provider: &v1alpha1.MeshConfig_ExtensionProvider_EnvoyExtAuthzHttp{
+			EnvoyExtAuthzHttp: &v1alpha1.MeshConfig_ExtensionProvider_EnvoyExternalAuthorizationHttpProvider{
+				Service:               service,
+				Port:                  8000,
+				PathPrefix:            "/check",
+				IncludeHeadersInCheck: []string{"x-ext-authz"},
+			},
+		},
+	}
+	grpcProvider := &v1alpha1.MeshConfig_ExtensionProvider{
+		Name: "ext-authz-grpc",
+		Provider: &v1alpha1.MeshConfig_ExtensionProvider_EnvoyExtAuthzGrpc{
+			EnvoyExtAuthzGrpc: &v1alpha1.MeshConfig_ExtensionProvider_EnvoyExternalAuthorizationGrpcProvider{
+				Service: service,
+				Port:    9000,
+			},
+		},
+	}
+
+	// Update the mesh config.
+	meshConfig.ExtensionProviders = []*v1alpha1.MeshConfig_ExtensionProvider{httpProvider, grpcProvider}
+	configMap.Data["mesh"], err = protomarshal.ToYAML(meshConfig)
+	if err != nil {
+		ctx.Fatalf("Failed to marshal mesh config: %v", err)
+	}
+	if _, err := ctx.Clusters().Default().CoreV1().ConfigMaps(systemNs.Name()).Update(
+		context.Background(), configMap, v1.UpdateOptions{}); err != nil {
+		ctx.Fatalf("Failed to update config map: %v", err)
+	}
+
+	// Restart istiod to make sure the change to istiod is ready for the following tests.
```

Comment:
Will this still be needed later? I thought we fixed this issue?

Reply:
I find this makes the test more stable and less flaky when running locally, it makes sure the meshconfig is up to date when later applying the authz policy.